test/extensions/filters/http/aws_lambda/config_test.cc

```diff
@@ -159,6 +188,32 @@ TEST(AwsLambdaFilterConfigTest, AsynchrnousPerRouteConfig) {
   EXPECT_EQ(InvocationMode::Asynchronous, filter_settings_ptr->invocationMode());
 }
 
+TEST(AwsLambdaFilterConfigTest, StatPrefixPerRouteConfig) {
+  const std::string yaml = R"EOF(
+  invoke_config:
+    arn: "arn:aws:lambda:region:424242:function:fun"
+    payload_passthrough: false
+    invocation_mode: asynchronous
+  stat_prefix: per_route
+  )EOF";
+
+  LambdaPerRouteConfig proto_config;
+  TestUtility::loadFromYamlAndValidate(yaml, proto_config);
+
+  testing::NiceMock<Server::Configuration::MockServerFactoryContext> context;
+  AwsLambdaFilterFactory factory;
+
+  auto route_specific_config_ptr = factory.createRouteSpecificFilterConfig(
+      proto_config, context, ProtobufMessage::getStrictValidationVisitor());
+  Http::MockFilterChainFactoryCallbacks filter_callbacks;
+  ASSERT_NE(route_specific_config_ptr, nullptr);
+  auto filter_settings_ptr = std::const_pointer_cast<FilterSettings>(
+      std::static_pointer_cast<const FilterSettings>(route_specific_config_ptr));
+  EXPECT_EQ("per_routeaws_lambda.server_error", filter_settings_ptr->stats().server_error_.name());
+  EXPECT_EQ("per_routeaws_lambda.upstream_rq_payload_size",
```

Comment:
this string does not seems to follow convention of <prefix>.<filter> I check it as upstream filter
```
cluster.jmo-lambda.cluster.jmo-lambdaaws_lambda.server_error: 0
cluster.jmo-lambda.cluster.jmo-lambdaaws_lambda.upstream_rq_payload_size: P0(nan,340) P25(nan,342.5) P50(nan,345) P75(nan,347.5) P90(nan,349) P95(nan,349.5) P99(nan,349.9) P99.5(nan,349.95) P99.9(nan,349.99) P100(nan,350)
```
this is legacy, and I check with aws_request_sign and it is doing the same. but not all other filters takes this approach.

Reply:
This is probably an issue when upstream prefix is built at the moment to create the filter